### PR TITLE
Support React

### PR DESCRIPTION
**Pull request by jimporter**
This PR adds support for React and building via WebPack. We should probably decide what (if anything) to do with the Makefile, since there's now a build step that needs to run. Also, I'd really like it if we had a better way of managing non-WebPacked files (i.e. the stuff in `dist/`), since I hate having to cram the build artifacts in there and add them to .gitignore. I'd much prefer out-of-tree builds.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -13,16 +13,30 @@
...
   "keywords": [
     "passwords",
     "crypto",
     "security",
     "addon"
   ],
+  "scripts": {
+    "run": "jpm run",
+    "test": "echo \"Error: no test specified\" && exit 1"
+  },
```
Comment (linuxwolf) [inline package.json]: There should be an entry for running webpack here, e.g.:

```
  "build": "webpack"
```

**Comment by devinreams** [comment]
For testing out our Project board: I believe this fixes #8 
(which would now auto-close when this merges)

**Comment by linuxwolf** [comment]
1. My preference is to replace the Makefile with npm run scripts.
2. Is there a reason not to add all of dist/ to `.gitignore`?  I realize it means extra copying, but it keeps the distribution consolidated.  This is where `watchman` helps.

File: .gitignore [hunk 2]
```diff
@@ -1,3 +1,4 @@
-node_modules/
+/node_modules/
+/site/
 /lockbox-*.xpi
-/site
+/dist/chrome/content/aboutPage.js
```
Comment (linuxwolf) [inline .gitignore]: why not all of `/dist/`?

File: dist/chrome/content/aboutPage.html [hunk 3]
```diff
@@ -5,8 +5,9 @@
 <html>
 <head>
   <title>Lockbox</title>
+  <script src="chrome://lockbox/content/aboutPage.js" defer></script>
```
Comment (devinreams) [inline dist/chrome/content/aboutPage.html]: Now that this is back to about:lockbox, is this path correct?

Comment (devinreams) [inline dist/chrome/content/aboutPage.html]: Follow-up: yes, it is. I just need to run webpack to actually get this asset in the right place.

**Comment by devinreams** [comment]
worksonmymachine

**Pull request merged**